src/main/java/edu/uci/ics/crawler4j/crawler/CrawlController.java

```diff
@@ -534,4 +551,69 @@ public void shutdown() {
         pageFetcher.shutDown();
         frontier.finish();
     }
+
+    /**
+     * Get the number of elapsed seconds since the crawl began or
+     * the total number of seconds the crawl took to complete
+     */
+    public long getCrawlElapsedSeconds() {
+        if (crawlLaunchedTimestamp == null) {
+            return 0;
+        }
+
+        if (crawlCompletedTimestamp != null) {
+            return crawlCompletedTimestamp.getEpochSecond() -
+                crawlLaunchedTimestamp.getEpochSecond();
+        } else {
+            return Instant.now().getEpochSecond() -
+                crawlLaunchedTimestamp.getEpochSecond();
+        }
+    }
+
+    /**
+     * Get the number of processed URLs
+     */
+    public long getProcessedUrlCount() {
+        return getFrontier().getNumberOfProcessedPages();
+    }
+
+    /**
+     * Get the total number of scheduled URLs
+     */
+    public long getScheduledUrlCount() {
+        return getFrontier().getNumberOfScheduledPages();
+    }
+
+    /**
+     * Get the percentage of URLs crawled
+     *
+     * WARNING: currently this is only accurate if you crawl *all* URLs
```

Comment:
Please don't add a stat which isn't applicable to all. If someone is interested in this it can be computed from processed and scheduled stats.